kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,207 @@
+    servers="(.spec.managedServers)"
+    serverStartPolicyPatch=$(echo ${domainJson} | jq "${replacePolicyCmd}" | jq -cr "${servers}")
+  fi
+  eval $__result="'${serverStartPolicyPatch}'"
+  eval $__currentPolicy=${currentServerStartPolicy}
+}
+
+#
+# Function to create patch string for updating replica count
+# $1 - Domain resource in json format
+# $2 - Name of cluster whose replica count will be patched
+# $3 - operatation string indicating whether to increment or decrement count
+# $4 - Return value containing replica update patch string
+# $5 - Retrun value containing updated replica count
+#
+function createReplicaPatch {
+local domainJson=$1
+local clusterName=$2
+local operation=$3
+local __result=$4
+local __relicaCount=$5
+maxReplicas=""
+local errorMessage="@@ ERROR: Maximum number of servers allowed (maxReplica = ${maxReplicas}) \
+are already running. Please increase cluster size to start new servers."
+
+  replicasCmd="(.spec.clusters[] \
+    | select (.clusterName == \"${clusterName}\")).replicas"
+  maxReplicaCmd="(.status.clusters[] | select (.clusterName == \"${clusterName}\")) \
+    | .maximumReplicas"
+  if [ "${operation}" == "DECREMENT" ]; then
+    replica=$(($(echo ${domainJson} | jq "${replicasCmd}")-1))
+    if [ ${replica} -lt 0 ]; then
+      replica=0
+    fi
+  elif [ "${operation}" == "INCREMENT" ]; then
+    replica=$(($(echo ${domainJson} | jq "${replicasCmd}")+1))
+    maxReplicas=$(echo ${domainJson} | jq "${maxReplicaCmd}")
+    if [ ${replica} -gt ${maxReplicas} ]; then
+      echo "${errorMessage}"
+      eval $__result="MAX_REPLICA_COUNT_EXCEEDED"
+      return
+    fi
+  fi
+
+  cmd="(.spec.clusters[] | select (.clusterName == \"${clusterName}\") \
+    | .replicas) |= ${replica}"
+  replicaPatch=$(echo ${domainJson} | jq "${cmd}" | jq -cr '(.spec.clusters)')
+  eval $__result="'${replicaPatch}'"
+  eval $__relicaCount="'${replica}'"
+}
+
+#
+# Function to validate whether a server belongs to a  cluster or is an independent managed server
+# $1 - Domain unique id.
+# $2 - Domain namespace.
+# $3 - Return value indicating if server is valid (i.e. if it's part of a cluster or independent server).
+# $4 - Retrun value containting cluster name to which this server belongs.
+#
+function validateServerAndFindCluster {
+local domainUid=$1
+local domainNamespace=$2 
+local __isValidServer=$3
+local __clusterName=$4
+local errorMessage="Server name is outside the range of allowed servers. \
+Please make sure server name is correct."
+
+  configMap=$(${kubernetesCli} get cm ${domainUid}-weblogic-domain-introspect-cm \
+    -n ${domainNamespace} -o json)
+  topology=$(echo "${configMap}" | jq '.data["topology.yaml"]')
+  jsonTopology=$(python -c \
+    'import sys, yaml, json; print json.dumps(yaml.safe_load('"${topology}"'), indent=4)')
+  servers=($(echo $jsonTopology | jq -r '.domain.servers[].name'))
+  if  checkStringInArray "${serverName}" "${servers[@]}" ; then
```

Comment:
Is it a requirement for this script to work using the (ancient) version of bash on MacOS? We have that requirement for other k8s scripts, but I hope not for this script.  But if so, then please do some ad hoc testing to make sure it works.  Note that jq isn't available on MacOS by default.

Reply:
Hi Tom, I'm not sure if MacOS is a requirement and I don't have an environment for ad hoc testing (my laptop is Windows laptop). In the JIRA requirement, it's mentioned that `jq` can be used and script needs to validate that `jq` is installed and fail if it's not installed.

Reply:
@mriccell @rjeberhard Thoughts?

Reply:
This needs to work on Mac. Customers aren't going to understand that it doesn't because the MacOS bash version is outdated :(.